source/src/util/cs_DoubleStackCoroutine.c

```diff
@@ -60,30 +61,36 @@ void startCoroutine(coroutine_t* coroutine, coroutineFunc coroutineFunction, voi
  */
 int nextCoroutine() {
 	// Save current context (registers) for bluenet.
-	coroutine_t* coroutine = stackParams->coroutine;
-	int ret                = setjmp(coroutine->bluenetContext);
+	int ret = setjmp(_stackParams->coroutine.bluenetContext);
 	if (ret == 0) {
-		// The bluenet registers are saved, fine now to jump to the microapp.
-		longjmp(coroutine->microapp0Context, WORKING);
+		// The first call to setjmp always returns 0.
+		// The bluenet registers are saved.
+		// Jump to the coroutine, we end up at the last setjmp with coroutine context, returning non 0.
+		longjmp(_stackParams->coroutine.coroutineContext, WORKING);
```

Comment:
Same remark as above. I strongly prefer `_stackParams` on a fixed position. This is gonna make it much harder if we are gonna add memory protection. I've also experimented with position-independent code for the microapp, which also is becoming harder when things are made more dynamic. There seems to be no good reason not to have this meta-information at a reserved spot just before the stack of the microapp.

Reply:
Misunderstood. It is at the end of ram.